src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -72,15 +72,26 @@ struct InternalFlashFactoryData
 
     constexpr size_t FactoryDataBlockSize()
     {
-        // calculate the factory data end address rounded up to the nearest multiple of CONFIG_FPROTECT_BLOCK_SIZE
+        // calculate the factory data and settings end address rounded up to the nearest multiple of CONFIG_FPROTECT_BLOCK_SIZE
         // and make sure we do not overlap with settings partition
         constexpr size_t kFactoryDataBlockEnd =
             (FACTORY_DATA_ADDRESS + FACTORY_DATA_SIZE + CONFIG_FPROTECT_BLOCK_SIZE - 1) & (-CONFIG_FPROTECT_BLOCK_SIZE);
-        static_assert(kFactoryDataBlockEnd <= PM_SETTINGS_STORAGE_ADDRESS,
+
+        constexpr size_t SettingsBlockEnd =
+            (PM_SETTINGS_STORAGE_ADDRESS + PM_SETTINGS_STORAGE_SIZE + CONFIG_FPROTECT_BLOCK_SIZE - 1) &
+            (-CONFIG_FPROTECT_BLOCK_SIZE);
+
+        constexpr size_t kFactoryDataBlockBegin = FACTORY_DATA_ADDRESS & (-CONFIG_FPROTECT_BLOCK_SIZE);
+
+        constexpr bool overlapsCheck =
+            (SettingsBlockEnd <= kFactoryDataBlockBegin) || (kFactoryDataBlockEnd <= PM_SETTINGS_STORAGE_ADDRESS);
```

Comment:
I'm not sure that this is correct. The problem that we consider here is the fact that fprotect has some fixed block size and we want to use only on factory data (not on settings). There are two issues possible:

1. The factory data is before the settings and its size is not aligned to the fprotect block size, so bigger range than expected will be covered by fprotect. In such case we have to check if factory data start+ factory data size + alignment to fprotect block size will not overlap with the start address of settings.
2. The settings are before the factory data and the factory data size is not aligned. If the factory data is at the end of memory, the fprotect will do the alignment using the space before the factory data (because after that there is no memory left). In such case we have to check if memory end - (factory data size + alignment) does not overlap with the settings end.